src/utils/configuration.h

```diff
@@ -43,11 +44,68 @@ namespace dsn {
 
 class configuration
 {
+private:
+    struct conf
+    {
+        std::string section;
+        std::string key;
+        std::string value;
+        int line;
+
+        bool present;
```

Comment:
Could you please add some comments when you happened to modify the code? Thanks!

Reply:
At begin I declaring `configuration(config_map&)` to use the mutex lock the map. If I not move private parameter before public function, an error will be reported during the compilation phase: the variable cannot be found.